src/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -117,10 +117,40 @@ public static int ByteLength(Array array)
             return _ByteLength(array);
         }
 
-        internal static unsafe void ZeroMemory(byte* src, long len)
+        // This is currently used by System.IO.UnmanagedMemoryStream
+        internal static unsafe void ZeroMemory(byte* dest, long len)
         {
-            while (len-- > 0)
-                *(src + len) = 0;
+#if BIT64
+            ZeroMemory(dest, (ulong)(len));
+#else
+            ulong length = (ulong)(len);
+
+            if (length <= nuint.MaxValue)
+            {
+                ZeroMemory(dest, (nuint)(length));
+            }
+            else
+            {
+                while (length >= nuint.MaxValue)
+                {
+                    length -= nuint.MaxValue;
+                    ZeroMemory(dest, nuint.MaxValue);
+                    dest += nuint.MaxValue;
+                }
+
+                if (length != 0)
+                {
+                    Debug.Assert(length < nuint.MaxValue);
+                    ZeroMemory(dest, (nuint)(length));
+                }
+            }
+#endif
+        }
+
+        // This method has different signature for x64 and other platforms and is done for performance reasons.
+        internal static unsafe void ZeroMemory(byte* dest, nuint len)
+        {
+            SpanHelpers.ClearWithoutReferences(ref Unsafe.AsRef<byte>(dest), (nuint)(len));
```

Comment:
Can this be just `SpanHelpers.ClearWithoutReferences(ref *dest, len)` ?

Reply:
Fixed.